snapcraft/parts/plugins/flutter.py

```diff
@@ -0,0 +1,64 @@
+from typing import List, Dict, Set, cast, Literal, Any
+
+from craft_parts import plugins
+
+FLUTTER_REPO = "https://github.com/flutter/flutter.git "
+
+class FlutterPluginProperties(plugins.PluginProperties, plugins.PluginModel):
+    flutter_branch: Literal["stable", "master", "dev"] = "stable"
+    flutter_target: str = "lib/main.dart"
+
+    @classmethod
+    def unmarshal(cls, data: Dict[str, Any]) -> "CondaPluginProperties":
```

Comment:
Comparing this to the colcon plugin, I'd think the unmarshal method should return "FlutterPluginProperties"

https://github.com/snapcore/snapcraft/blob/416c76337cd3da3cab87c856a66de111d351d5b5/snapcraft/parts/plugins/colcon.py#L78-L80